src/foam/box/SessionServerBox.java

```diff
@@ -94,7 +94,8 @@ public void send(Message msg) {
       session.touch();
 
       // TODO: Shouldn't this go somewhere else?
-      if ( req != null && ! SafetyUtil.isEmpty(req.getRequestURI()) ) {
+      if ( effectiveContext.get("group") == null &&
```

Comment:
How about some comments to explain what this is doing and why?

Reply:
Moved to Session where it's obvious.